src/meta/src/hummock/compaction_scheduler.rs

```diff
@@ -459,21 +451,11 @@ where
         task_type: compact_task::TaskType,
         compaction_selectors: &mut HashMap<compact_task::TaskType, Box<dyn LevelSelector>>,
         sched_channel: Arc<CompactionRequestChannel>,
-        mut shutdown_rx: Shared<Receiver<()>>,
     ) -> bool {
         // Wait for a compactor to become available.
-        let compactor = loop {
-            if let Some(compactor) = self.hummock_manager.get_idle_compactor().await {
-                break compactor;
-            } else {
-                tracing::debug!("No available compactor, pausing compaction.");
-                tokio::select! {
-                    _ = self.compaction_resume_notifier.notified() => {},
-                    _ = &mut shutdown_rx => {
-                        return false;
-                    }
-                }
-            }
+        let compactor = match self.hummock_manager.get_idle_compactor().await {
+            Some(compactor) => compactor,
+            None => return false,
```

Comment:
This will cause the schedule loop to break and exit: https://github.com/risingwavelabs/risingwave/blob/54052caa68cabec3cda29426704ae00d2038a55d/src/meta/src/hummock/compaction_scheduler.rs#L348